cmd/clusterctl/pkg/client/client.go

```diff
@@ -129,11 +140,17 @@ func newClusterctlClient(path string, options ...Option) (*clusterctlClient, err
 		o(cfg)
 	}
 
+	// if there is an injected logger, use it, otherwise use a default one
+	logger := cfg.injectLogger
+	if logger == nil {
+		logger = klogr.New() //TODO: replace with a logger with a better output
```

Comment:
Any specific reason to not just return an error here instead? That would reduce the dependency on klogr here.

If the goal is to have a common "friendly" logger that folks can use, then couldn't that be offered as a separate package they could import to pass in to New here?

Reply:
I opened https://github.com/kubernetes-sigs/cluster-api/issues/2164 to tackle this problem

Reply:
I think it would be ideal if we don't force library consumers to import `klog` or `klogr` unless they actually want to.